test/http-client-tests/src/main/java/software/amazon/awssdk/http/proxy/HttpClientDefaultPoxyConfigTest.java

```diff
@@ -0,0 +1,300 @@
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import software.amazon.awssdk.http.EmptyPublisher;
+import software.amazon.awssdk.http.ExecutableHttpRequest;
+import software.amazon.awssdk.http.HttpExecuteRequest;
+import software.amazon.awssdk.http.HttpExecuteResponse;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.http.SdkHttpResponse;
+import software.amazon.awssdk.http.async.AsyncExecuteRequest;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.http.async.SdkAsyncHttpResponseHandler;
+import software.amazon.awssdk.testutils.EnvironmentVariableHelper;
+import software.amazon.awssdk.utils.Pair;
+
+public abstract class HttpClientDefaultPoxyConfigTest {
+
+    static Random random = new Random();
+
+    private static final EnvironmentVariableHelper ENVIRONMENT_VARIABLE_HELPER = new EnvironmentVariableHelper();
+
+    protected WireMockServer mockProxy = new WireMockServer(new WireMockConfiguration()
+                                                                .dynamicPort()
+                                                                .dynamicHttpsPort()
+                                                                .enableBrowserProxying(true));
+
+    protected WireMockServer mockServer = new WireMockServer(new WireMockConfiguration()
+                                                                 .dynamicPort()
+                                                                 .dynamicHttpsPort());
+
+    protected abstract boolean isSyncClient();
+
+    protected abstract SdkAsyncHttpClient createHttpClientWithDefaultProxy();
+
+    protected abstract SdkHttpClient createSyncHttpClientWithDefaultProxy();
+
+    protected abstract Class<? extends Exception> getProxyFailedExceptionType();
+
+    protected abstract Class<? extends Exception> getProxyFailedCauseExceptionType();
+
+
+    @BeforeEach
+    public void setup() {
+        mockProxy.start();
+        mockServer.start();
+        mockServer.stubFor(get(WireMock.urlMatching(".*"))
+                               .willReturn(aResponse().withStatus(200).withBody("hello")));
+    }
+
+    @AfterEach
+    public void teardown() {
+        mockServer.stop();
+        mockProxy.stop();
+        ENVIRONMENT_VARIABLE_HELPER.reset();
+        System.clearProperty("http.proxyHost");
+        System.clearProperty("http.proxyPort");
+    }
+
+    public static Stream<Arguments> proxyConfigurationSettingWithConnectionFails() {
+        return Stream.of(
+            Arguments.of(
+                Arrays.asList(
+                    Pair.of("http.proxyHost", "localhost"),
```

Comment:
Seems like these are supposed to be environment variables right (we're using `ENVIRONMENT_VARIABLE_HELPER`)? Shouldn't it be `HTTP_PROXY` or something similar?

Reply:
This is Systen property , the one below it is Environment variable